lib/llm/provider.rb

```diff
@@ -1,13 +1,17 @@
 # frozen_string_literal: true
 
 module LLM
-  # The Provider class represents a base class for LLM (Language Model) providers.
+  ##
+  # The Provider class represents a base class for
```

Comment:
IMO this could be `abstract class`. Thoughts @antaz ?

Reply:
@0x1eef Yes! I agree.